### ISSUE TITLE
how does masked_lm_labels  work ?

### ISSUE THREAD

**Issue by mahdirezaey** [huggingface/transformers#3315]
# ❓ Questions & Help

## Details

**A link to original question on Stack Overflow**:

**Comment by mahdirezaey** [comment]
Hi all
from hugging face{https://huggingface.co/transformers/model_doc/bert.html#bertformaskedlm}

in this code :
{
from transformers import BertTokenizer, BertForMaskedLM
import torch

tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
model = BertForMaskedLM.from_pretrained('bert-base-uncased')

input_ids = torch.tensor(tokenizer.encode("Hello, my dog is cute", add_special_tokens=True)).unsqueeze(0) # Batch size 1
outputs = model(input_ids, masked_lm_labels=input_ids)

loss, prediction_scores = outputs[:2]
}

what happens by {outputs = model(input_ids, masked_lm_labels=input_ids)} ?

it will automatically makes [mask] 15% of all the tokens in each sentence of each bs , and calculates loss just for them ?
@thomwolf @tholor

**Comment by blackcat84** [comment]
No I don't think so, you need to mask the tokens and then pass them to model, look here (e.g. evaluate) for an example
https://github.com/huggingface/transformers/blob/master/examples/run_language_modeling.py

**Comment by mahdirezaey** [comment]
@blackcat84 , thanks it helped a lot

**Comment by mahdirezaey** [comment]
@blackcat84 

and one more thing ,
does any function in those scripts , concatenate the short lines to each other ?
in order not to be enforced to pad each line so much

**Comment by blackcat84** [comment]
It's been a while so I might be wrong but I think you are correct, in the function that prepares the batches. A simple way to be sure about it is to pass a dummy input to the script/function and check it by yourself
